src/main/scala/loamstream/loam/intake/aggregator/SourceColumns.scala

```diff
@@ -5,32 +5,57 @@ import loamstream.loam.intake.ColumnName
 final case class SourceColumns(
     marker: ColumnName,
     pValue: ColumnName,
-    zScore: ColumnName,
-    stderr: ColumnName,
-    beta: ColumnName,
-    oddsRatio: ColumnName,
-    eaf: ColumnName,
-    maf: ColumnName) {
+    zScore: Option[ColumnName] = None,
+    stderr: Option[ColumnName] = None,
+    beta: Option[ColumnName] = None,
+    oddsRatio: Option[ColumnName] = None,
+    eaf: Option[ColumnName] = None,
+    maf: Option[ColumnName] = None) {
   
-  private val mapping: Map[ColumnName, ColumnName] = Map(
+  validate()
+  
+  private val mapping: Map[ColumnName, ColumnName] = {
+    //mandatory columns
+    val mandatory = Map(
       ColumnNames.marker -> this.marker,
-      ColumnNames.pvalue -> this.pValue,
-      ColumnNames.zscore -> this.zScore,
-      ColumnNames.stderr -> this.stderr,
-      ColumnNames.beta -> this.beta,
-      ColumnNames.odds_ratio -> this.oddsRatio,
-      ColumnNames.eaf -> this.eaf,
-      ColumnNames.maf -> this.maf)
+      ColumnNames.pvalue -> this.pValue)
       
-  def asConfigFileContents: String = {
-    val aggregatorColumnNameToSourceColumnNameLines = for {
-      aggregatorColumnName <- ColumnNames.values
-    } yield {
-      val sourceColumnName = mapping(aggregatorColumnName)
+    mandatory ++
+      stderr.map(ColumnNames.stderr -> _) ++
+      zScore.map(ColumnNames.zscore -> _) ++
+      beta.map(ColumnNames.beta -> _) ++
+      oddsRatio.map(ColumnNames.odds_ratio -> _) ++
+      eaf.map(ColumnNames.eaf -> _) ++
+      maf.map(ColumnNames.maf -> _)
+  }
       
-      s"${aggregatorColumnName.name} ${sourceColumnName.name}"
+  def asConfigFileContents: String = {
+    val aggregatorColumnNameToSourceColumnNameLines = mapping.map {
+      case (aggregatorColumnName, sourceColumnName) => s"${aggregatorColumnName.name} ${sourceColumnName.name}"
     }
     
     aggregatorColumnNameToSourceColumnNameLines.mkString(System.lineSeparator)
   }
+  
+  private def validate(): Unit = {
+    if(maf.isEmpty) {
+      require(eaf.isDefined)
+    }
+    
+    if(beta.isEmpty) {
+      require(oddsRatio.isDefined)
+    }
+    
+    if(oddsRatio.isEmpty) {
+      require(beta.isDefined)
+    }
+    
+    if(stderr.isEmpty) {
+      require(beta.isDefined)
+    }
+    
+    if(zScore.isEmpty) {
+      require(beta.isDefined)
+    }
```

Comment:
zScore requires both beta and standard error.

Reply:
Thank you!